lib/src/signed_video_openssl.c

```diff
@@ -90,33 +90,46 @@ openssl_create_private_key(sign_algo_t algo, const char *path_to_key, key_paths_
 #define PRIVATE_RSA_KEY_FILE "private_rsa_key.pem"
 #define PRIVATE_ECDSA_KEY_FILE "private_ecdsa_key.pem"
 
-/* Allocates enough memory for the signature when using the |private_key| in |signature_info|. */
+/* Frees an EVP_PKEY object. */
+void
+openssl_free_key(void *pkey)
+{
+  EVP_PKEY_free((EVP_PKEY *)pkey);
+}
+
+/* Reads the |private_key| which is expected to be on PEM form and creates an EVP_PKEY
+ * object out of it and sets it in |signature_info|. Further, enough memory for the signature
+ * is allocated. */
 SignedVideoReturnCode
-openssl_signature_malloc(signature_info_t *signature_info)
+openssl_private_key_malloc(signature_info_t *signature_info,
+    const char *private_key,
+    size_t private_key_size)
 {
   // Sanity check input
-  if (!signature_info) return SV_INVALID_PARAMETER;
+  if (!signature_info || !private_key || private_key_size == 0) return SV_INVALID_PARAMETER;
 
-  const void *private_key = signature_info->private_key;
-  int private_key_size = (int)signature_info->private_key_size;
+  EVP_PKEY *signing_key = NULL;
   svi_rc status = SVI_UNKNOWN;
   SVI_TRY()
-    SVI_THROW_IF(!private_key || private_key_size == 0, SV_INVALID_PARAMETER);
+    SVI_THROW_IF(!private_key, SVI_INVALID_PARAMETER);
```

Comment:
Check of "!private_key" is already performed at line 109.

Reply:
Correct, and it is actually removed in another PR. I'll fix it in this PR to make this one more correct